src/Features/CSharp/Portable/Completion/KeywordRecommenders/StructKeywordRecommender.cs

```diff
@@ -39,6 +39,34 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 canBePartial: true,
                 cancellationToken: cancellationToken) ||
             context.IsRecordDeclarationContext(s_validModifiers, cancellationToken) ||
-            syntaxTree.IsTypeParameterConstraintStartContext(position, context.LeftToken);
+            IsAllowsRefStructConstraintContext(context);
+    }
+
+    private static bool IsAllowsRefStructConstraintContext(CSharpSyntaxContext context)
+    {
+        //    where T : |
+        if (context.SyntaxTree.IsTypeParameterConstraintStartContext(context.Position, context.LeftToken))
```

Comment:
I wouldn't expect this check to be inside method `IsAllowsRefStructConstraintContext`, consider returning it where it was previously.

Reply:
I'll rename the method